File: ios/CrashTester.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CrashTester, NSObject)

RCT_EXTERN_METHOD(nativeCrash:(NSString *)message)

RCT_EXPORT_METHOD(nativeException:(NSString *)message) {
  NSLog(@"Raising test exception");
  [NSException raise:@"Native Exception" format:@"%@", [message description]];
}

@end

File: ios/CrashTester.swift
import Foundation

@objc(CrashTester)
class CrashTester: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc(nativeCrash:)
  func nativeCrash(_ message: String = "") -> Void {
      fatalError(message)
  }
}
